go/mysql/binlog_event_compression.go

```diff
@@ -292,6 +297,8 @@ func (tp *TransactionPayload) decompress() error {
 		}
 		compressedTrxPayloadsUsingStream.Add(1)
 		tp.reader = streamDecoder
+		// Signal the consumer to also stream the contents.
+		tp.StreamingContents = true
```

Comment:
Should we just always stream? (Meaning we would remove the conditional flag?)

I guess there's also a backwards-compatibility issue here?

Reply:
I thought about that too. It's more efficient to do it in a batch though so for smaller payloads we do that (one batch) at both layers, binlog and vstreamer.